core/src/main/scala/io/projectglow/vcf/VariantContextToInternalRowConverter.scala

```diff
@@ -594,18 +595,32 @@ class VariantContextToInternalRowConverter(
     case other: Any => converter(parseObjectAsString(other))
   }
 
-  private def obj2array[T <: AnyRef: ClassTag, R <: AnyVal](
-      converter: String => T)(obj: Object, primitiveConverter: Option[R => T] = None): Array[Any] =
+  private def obj2array[T <: AnyRef, R <: AnyVal](converter: String => T)(
+      obj: Object,
+      primitiveConverter: Option[R => T] = None)(implicit ct: ClassTag[T]): Array[Any] =
     obj match {
       case null => null
       case VCFConstants.MISSING_VALUE_v4 => null
-      case arr: Array[T] => arr.asInstanceOf[Array[Any]]
-      case arr: Array[R] if primitiveConverter.isDefined => arr.map(primitiveConverter.get)
-      case l: JList[T] =>
+      case arr: Array[AnyRef] =>
+        var i = 0
+        while (i < arr.length) {
```

Comment:
Do we have to check for every element, or is just the head enough?

Reply:
It is possible for each element to have a different type if the original collection type was like `Array[Object]`. Knowing htsjdk, I wouldn't count against that.

Reply:
OOC, does this add to the runtime significantly?